cmd/argoexec/commands/emissary.go

```diff
@@ -203,12 +211,60 @@ func NewEmissaryCommand() *cobra.Command {
 			} else {
 				logger.Info("not saving outputs - not main container")
 			}
-
 			return cmdErr // this is the error returned from cmd.Wait(), which maybe an exitError
 		},
 	}
 }
 
+func getBackoff(containerset *wfv1.ContainerSetTemplate) (wait.Backoff, error) {
```

Comment:
I think this command could be changed to be encapsulated into:

```golang
func (t *Template) GetRetryStrategy() (*wait.Backoff,error) {
  return t.ContainerSet.GetRetryStrategy()
}

func (t * ContainerSetTemplate) GetRetryStrategy() (*wait.Backoff,error) {
  if t == nil {
    return &wait.Backoff{Steps: 1}, nil
  }
// ...
```

Then you can add unit tests for it that are de-coupled from this package.

Reply:
Good point! Will update it.